src/Common/PSR16Util.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Vectorface\Cache\Common;
+
+use DateInterval;
+use DateTime;
+use Traversable;
+use Vectorface\Cache\Exception\CacheException;
+use Vectorface\Cache\Exception\InvalidArgumentException as CacheArgumentException;
+
+
+/**
+ *
+ */
+trait PSR16Util
+{
+    /**
+     * The DateTime implementation to use
+     *
+     * @var string
+     */
+    private static $dateTimeClass = DateTime::class;
+
+    /**
+     * Enforce a fairly standard key format
+     *
+     * @param $key
+     * @return mixed Returns the key, if valid
+     * @throws CacheArgumentException Thrown if the key is not a legal value
+     */
+    protected function key($key)
+    {
+        if (is_scalar($key)) {
+            return $key;
+        }
+
+        throw new CacheArgumentException("key is not a legal value");
+    }
+
+    /**
+     * Enforce fairly standard key formats on an iterable of values
+     * @param iterable $values
+     * @return iterable The values array
+     * @throws CacheArgumentException Thrown if any of the keys is not a legal value
+     */
+    protected function values($values) {
+        foreach ($values as $key => $value) {
+            $key = $this->key($key); // Checks the key
+        }
+        return $values;
+    }
+
+    /**
+     * Enforce a fairly standard key format on an array or Traversable of keys
+     *
+     * @param iterable $keys
+     * @return mixed Returns the key, if valid
```

Comment:
```suggestion
     * @return mixed[] Returns the keys, if valid
```

or, but a bit silly:

```suggestion
     * @return int[]|float[]|string[]|bool[] Returns the keys, if valid
```

Reply:
What I want is `scalar[]`! 😁

Reply:
Actually, what if we just required `is_numeric() || is_string()` instead of `is_scalar()`? I don't think booleans make much sense as a cache key.